app/src/main/res/layout/custom_barcode_scanner.xml

```diff
@@ -19,23 +19,40 @@
         app:zxing_viewfinder_laser="@color/zxing_custom_viewfinder_laser"
         app:zxing_viewfinder_mask="@color/zxing_custom_viewfinder_mask"/>
 
-    <LinearLayout
+    <androidx.constraintlayout.widget.ConstraintLayout
         android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:layout_gravity="bottom|center_horizontal"
-        android:gravity="center_horizontal"
-        android:orientation="vertical">
+        android:layout_height="match_parent"
+        android:padding="@dimen/activity_scanner_padding"
+        android:layout_gravity="bottom|center_horizontal">
+
+        <TextView
+            android:visibility="gone"
+            android:id="@+id/tap_give_permission"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="@string/noCameraPermissionDirectToSystemSetting"
+            android:textColor="@android:color/white"
+            app:layout_constraintBottom_toBottomOf="@+id/add_manually"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
```

Comment:
Hmm, I don't really like the design of this much. Would it be possible to make it look more like the home empty state screen? You could copy the layout from `main_activity.xml` more likely. I think using `no_photography` as icon (should be available in Android Studio's "Vector Asset" picker) would work well.

![image](https://user-images.githubusercontent.com/1885159/195917003-fb19bf77-2521-412a-ad46-329400575fa6.png)

Reply:
Sure, no problem, I’ll fix the stuff you mentioned.

Reply:
Does this look better?
![Screenshot_20221015_105002](https://user-images.githubusercontent.com/8223139/195992817-9616a664-3089-4ac8-af30-7d8840e76ec4.png)

Reply:
That looks awesome! Slight grammar issue though, should be "We can't access the camera"

Reply:
Got it, grammar issue is now fixed.

I have pushed a new commit [c14ece2](https://github.com/CatimaLoyalty/Android/pull/1085/commits/c14ece2173b7c64490057130be7798eb0504ce2d) to reflect my changes.